apis/core/v1alpha1/condition.go

```diff
@@ -53,6 +56,13 @@ const (
 	ReasonReconcileError   ConditionReason = "Encountered an error during managed resource reconciliation"
 )
 
+// Reason references for a resource are or are not resolved
+const (
+	ReasonReferenceResolveSuccess  ConditionReason = "Successfully resolved managed resource references to other resources"
+	ReasonResolveReferencesBlocked ConditionReason = "One or more of referred resources do not exists, or are not yet Ready"
+	ReasonReferenceResolveError    ConditionReason = "Encountered an error during resolving managed resource references to other resources"
+)
```

Comment:
Are you suggesting to use the existing `TypeReady`, instead of creating creating the new `TypeReferencesResolved` ConditionType? If so I'd slightly prefer to create a new type to avoid modifying the existing code. Also, a separate condition would tell the history of reference resolution in resource rather than re-writing it with the most up to date `Ready` condition.

Reply:
I can see your point, and I agree that if we use `TypeReady`, `ReferencesResolved` status can be deduced by,  say `Ready`. However having a separate condition enables us to skip running `ResolveReferences` in each reconcile step, when `ReferencesResolved` status is overwritten by a later `ReconcileError` status.